u2f-ref-code/java/tests/com/google/u2f/server/ControlFlagsTest.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * 
+ */
+package com.google.u2f.server;
+
+import static org.junit.Assert.*;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author at
+ *
+ */
+public class ControlFlagsTest {
+  protected byte userPresent, transferAccessMessage, both, neither;
+
+  @Before
+  public void setUp() throws Exception {
+    userPresent = 0x1;
+    transferAccessMessage = 0x2;
+    both = 0x3;
+    neither = 0x0;
+  }
+
+  /**
+   * @throws java.lang.Exception
+   */
+  @After
+  public void tearDown() throws Exception {}
+
+  @Test
+  public void testToByteFromByte() {
```

Comment:
this is a good test to do, but I would also have a single byte that you try to parse.  Because it's conceivable that without doing so, you hide bugs.  Also, diagnosing bugs will be harder the more methods you call.